packages/plugin-directives/src/mock-ast.ts

```diff
@@ -0,0 +1,240 @@
+      };
+    } else if (type instanceof GraphQLInterfaceType) {
+      type.astNode = {
+        kind: 'InterfaceTypeDefinition',
+        name: { kind: 'Name', value: typeName },
+        description: type.description
+          ? { kind: 'StringValue', value: type.description }
+          : undefined,
+        interfaces: type.getInterfaces().map((iface) => typeNode(iface) as NamedTypeNode),
+        fields: fieldNodes(type.getFields()),
+        directives: type.extensions?.directives
+          ? directiveNodes(type.extensions.directives as DirectiveList)
+          : [],
+      };
+    } else if (type instanceof GraphQLUnionType) {
+      type.astNode = {
+        kind: 'UnionTypeDefinition',
+        name: { kind: 'Name', value: typeName },
+        description: type.description
+          ? { kind: 'StringValue', value: type.description }
+          : undefined,
+        types: type.getTypes().map((iface) => typeNode(iface) as NamedTypeNode),
+        directives: type.extensions?.directives
+          ? directiveNodes(type.extensions.directives as DirectiveList)
+          : [],
+      };
+    } else if (type instanceof GraphQLEnumType) {
+      type.astNode = {
+        kind: 'EnumTypeDefinition',
+        name: { kind: 'Name', value: typeName },
+        description: type.description
+          ? { kind: 'StringValue', value: type.description }
+          : undefined,
+        values: enumValueNodes(type.getValues()),
+        directives: type.extensions?.directives
+          ? directiveNodes(type.extensions.directives as DirectiveList)
+          : [],
+      };
+    } else if (type instanceof GraphQLScalarType) {
+      type.astNode = {
+        kind: 'ScalarTypeDefinition',
+        name: { kind: 'Name', value: typeName },
+        description: type.description
+          ? { kind: 'StringValue', value: type.description }
+          : undefined,
+        directives: type.extensions?.directives
+          ? directiveNodes(type.extensions.directives as DirectiveList)
+          : [],
+      };
+    } else if (type instanceof GraphQLInputObjectType) {
+      type.astNode = {
+        kind: 'InputObjectTypeDefinition',
+        name: { kind: 'Name', value: typeName },
+        description: type.description
+          ? { kind: 'StringValue', value: type.description }
+          : undefined,
+        fields: inputFieldNodes(type.getFields()),
+        directives: type.extensions?.directives
+          ? directiveNodes(type.extensions.directives as DirectiveList)
+          : [],
+      };
+    }
+  });
+}
+
+function typeNode(type: GraphQLType): TypeNode {
+  if (type instanceof GraphQLList) {
+    return { kind: 'ListType', type: typeNode(type.ofType) };
+  }
+
+  if (type instanceof GraphQLNonNull) {
+    return { kind: 'NonNullType', type: typeNode(type.ofType) as NamedTypeNode | ListTypeNode };
+  }
+
+  return { kind: 'NamedType', name: { kind: 'Name', value: type.name } };
+}
+
+function valueNode(value: unknown): ValueNode {
```

Comment:
Need to document this hacky solution.  May be better to have all values be nulls.